pulp_container/tests/functional/conftest.py

```diff
@@ -30,10 +32,75 @@
 )
 
 
+class RegistryClient:
+    """A container registry client on a test runner machine."""
+
+    NAME = "podman"
+
+    def __init__(self):
+        self._name = None
+
+    @property
+    def name(self):
+        if not self._name:
+            self._name = self._get_registry_client()
+        return self._name
+
+    def raise_if_unsupported(self, exc, message="Unsupported registry client"):
+        try:
+            self.name
+        except RuntimeError:
+            raise exc(message)
+
+    @contextmanager
+    def set_env(self, **environ):
+        old_environ = os.environ.copy()
+        os.environ.update(environ)
+        try:
+            yield
+        finally:
```

Comment:
mm except missing?

Reply:
Not really. The goal is the clear up the updated environment variables at all costs.

Reply:
tbh i have never seen try block without except, got any examples?

Reply:
https://docs.python.org/3/reference/compound_stmts.html#finally-clause

Reply:
https://github.com/pulp/pulpcore/blob/48d0ef772af1b1484799bd86ba29979a0949be52/pulpcore/app/models/storage.py#L56-L59

Reply:
@mdellweg, do you see any added value in logging a raised error? Or, do you think it is sufficient to throw the exception to stdout/stderr as it is right now - no `except` statement?